tests/functional/ProvisionDispatcher.spec.js

```diff
@@ -44,12 +50,48 @@ function testDispatch() {
     });
 
     it('should be given all provisions when alone', done => {
-        clients[0] = new ProvisionDispatcher(zkConf);
+        const cbOnce = jsutil.once(done);
+        clients[0] = new ProvisionDispatcher(
+            zkConf, zk);
         clients[0].subscribe((err, items) => {
             assert.ifError(err);
             assert.deepStrictEqual(items, provisionList);
-            done();
-        });
+            cbOnce();
+        }, false);
+    });
+
+    it('should recheck if missing a watcher event', done => {
+        const cbOnce = jsutil.once(done);
+        clients[0] = new ProvisionDispatcher(
+            zkConf, zk);
+        let times = 0;
+        clients[0].subscribe((err, items) => {
+            assert.ifError(err);
+            if (times === 0) {
+                // first time we shall have all provisions
+                assert(items.length === 8);
+                // simulate a watcher event loss
+                const myPath =
+                      clients[0]._client._basePath +
+                      clients[0]._getMyPath();
+                const result = clients[0]._client._getZNode(myPath);
+                assert.ifError(result.err);
+                result.parent.children[result.baseName].emitter.removeAllListeners();
+                // introduce a new client
+                clients[1] = new ProvisionDispatcher(
+                    zkConf, zk);
+                clients[1].subscribe(err => {
+                    assert.ifError(err);
+                }, false);
+            } else {
+                // there might be intermediate callbacks but after the timer
+                // we shall get half of it
+                if (items.length === 4) {
```

Comment:
I'd suggest adding the explanation of why we get 4 items (I assume this is because there are now 2 clients that get an equal number of provisions each, but was not obvious for me).